components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/action/execution/PreIssueAccessTokenResponseProcessor.java

```diff
@@ -95,6 +99,13 @@ public ActionExecutionStatus<Success> processSuccessResponse(FlowContext flowCon
 
         AccessToken requestAccessToken = preIssueAccessTokenEvent.getAccessToken();
         AccessToken.Builder responseAccessTokenBuilder = preIssueAccessTokenEvent.getAccessToken().copy();
+
+        RefreshToken requestRefreshToken = preIssueAccessTokenEvent.getRefreshToken();
+        RefreshToken.Builder responseRefreshTokenBuilder = null;
+        if (requestRefreshToken != null) {
+            responseRefreshTokenBuilder = preIssueAccessTokenEvent.getRefreshToken().copy();
+        }
```

Comment:
Looks like through out we are checking if response token builder is null. Can't we use optional and improve readability

Reply:
Fixed 49caebcb5cca897a855b8ca05cb5db3e23952e9d